src/main/java/net/imglib2/type/PrimitiveType.java

```diff
@@ -0,0 +1,27 @@
+package net.imglib2.type;
+
+import net.imglib2.img.basictypeaccess.AccessFlags;
+import net.imglib2.img.basictypeaccess.array.ArrayDataAccess;
+import net.imglib2.img.basictypeaccess.volatiles.array.DirtyVolatileByteArray;
+
+/**
+ * Enumeration of Java primitive types which can back {@link NativeType}s.
+ * <p>
+ * In conjunction with {@link AccessFlags} this describes a specific
+ * {@link ArrayDataAccess}. For example, {@code BYTE} with flags {@code DIRTY}
+ * and {@code VOLATILE} specifies {@link DirtyVolatileByteArray}.
+ * </p>
+ *
+ * @author Tobias Pietzsch
+ */
+public enum PrimitiveType
+{
```

Comment:
`PrimitiveType` is a misleading name. (Rhetorical question: Why is there no `BOOLEAN`) The name `PrimitiveType` is not telling the truth. The class is intended to represent: `ByteAccess`, `CharAccess`, `ShortAccess`, `IntAccess`, ...

Conclusion: `AccessType` of `PrimitiveAccessType` would be a better name. The constants should be named `BYTE_ACCESS`, `CHAR_ACCESS`, `SHORT_ACCESS` ...

Overall. **This class is not really needed**: `ByteAccess.class`, `CharAccess.class`, ... could be used instead. This would make the intension of the code easier to understand.